src/pyfmi/simulation/assimulo_interface_fmi3.pyx

```diff
@@ -0,0 +1,611 @@
+        # cdef int enter_event_mode = 0, terminate_simulation = 0
+
+        # if self._extra_f_nbr > 0:
+        #     y_extra = solver.y[-self._extra_f_nbr:]
+        #     y       = solver.y[:-self._extra_f_nbr]
+        # else:
+        #     y       = solver.y
+
+        # # Moving data to the model
+        # if self._compare(solver.t, y):
+        #     self._update_model(solver.t, y)
+
+        #     # Evaluating the rhs (Have to evaluate the values in the model)
+        #     if self.model_me3_instance:
+        #         self.model_me3._get_derivatives(self._state_temp_1)
+        #     else:
+        #         rhs = self._model.get_derivatives()
+
+        # if self._logging_as_dynamic_diagnostics:
+        #     diag_data = np.ndarray(self._number_of_diagnostics_variables, dtype=float)
+        #     index = 0
+        #     diag_data[index] = solver.t
+        #     index +=1
+        #     if solver.clock_step:
+        #         diag_data[index] = solver.get_elapsed_step_time()
+        #         index +=1
+        #     solver_name = solver.__class__.__name__
+        #     if solver_name == "CVode":
+        #         diag_data[index] = solver.get_last_order()
+        #         index +=1
+        #     support_state_errors = (solver_name=="CVode" or solver_name=="Radau5ODE")
+        #     if support_state_errors and self._f_nbr > 0:
+        #         for e in solver.get_weighted_local_errors():
+        #             diag_data[index] = e
+        #             index +=1
+        #     support_event_indicators = (solver_name=="CVode" or solver_name=="Radau5ODE" or solver_name=="LSODAR" or solver_name=="ImplicitEuler")
+        #     if self._g_nbr > 0 and support_event_indicators:
+        #         for ei in ev_indicator_values:
+        #             diag_data[index] = ei
+        #             index +=1
+
+        #         for ei in range(len(ev_indicator_values)):
+        #             diag_data[index] = 0
+        #             index +=1
+        #     if support_event_indicators:
+        #         diag_data[index] = float(-1)
+        #     self.export.diagnostics_point(diag_data)
+
+
+        # if self.model_me3_instance:
+        #     self.model_me3._completed_integrator_step(&enter_event_mode, &terminate_simulation)
+        # else:
+        #     enter_event_mode, terminate_simulation = self._model.completed_integrator_step()
+        
+        # ret_flag = 0
+        # if enter_event_mode:
+        #     self._logg_step_event += [solver.t]
+        #     # Event have been detect, call event iteration.
+        #     self.handle_event(solver,[0])
+        #     ret_flag =  1 # Tell to reinitiate the solver.
+        
+        # if self._synchronize_factor > 0:
+        #     under_run = solver.t/self._synchronize_factor - (timer()-self._start_time)
+        #     if under_run > 0:
+        #         time.sleep(under_run)
+        
+        # return ret_flag
+
+    def _get_debug_file_object(self):
```

Comment:
Also this should likely be renamed to `get_and_open*` or similar, since the name of the function doesnt state that we actually `open` a resource here (which I would argue is vastly different).

Reply:
Actually, this function(ality) was purely used for the dynamic_diagnostic logging to XML; since we removed that, I could also cleanup this part.

Good that you probed this!